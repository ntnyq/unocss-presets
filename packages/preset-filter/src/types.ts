export interface Filter {
  /**
   * filter name
   */
  name: string

  /**
   * filter style
   */
  style: string

  /**
   * pseudo element before style
   */
  beforeStyle?: string
}

export interface PresetFilterOptions {
  /**
   * customize filter selector prefix
   *
   * @default 'filter-'
   */
  prefix?: string
}
