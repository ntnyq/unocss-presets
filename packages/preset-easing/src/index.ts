import { camelCase, kebabCase } from '@ntnyq/utils'
import { definePreset } from '@unocss/core'
import { easings } from './easings'
import type { PresetEasingOptions } from './types'

export const presetEasing = definePreset<PresetEasingOptions>((options = {}) => {
  const { prefix = 'easing-' } = options
  const selectorPattern = `${prefix}(${Object.keys(easings)
    .map(name => kebabCase(name))
    .join('|')})`

  return {
    name: 'unocss-preset-easing',

    rules: [
      [
        new RegExp(`^${selectorPattern}$`),
        ([, name]) => ({
          'transition-timing-function': easings[camelCase(name) as keyof typeof easings],
        }),
        {
          autocomplete: [selectorPattern],
        },
      ],
    ],
  }
})

export * from './types'
export * from './easings'

export default presetEasing
