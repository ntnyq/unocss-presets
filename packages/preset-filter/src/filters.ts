import type { Filter } from './types'

export function defineFilter(filter: Filter) {
  return filter
}

export const filters = [
  defineFilter({
    name: '1977',
    style: 'filter: sepia(.5) hue-rotate(-30deg) saturate(1.4)',
  }),

  defineFilter({
    name: 'aden',
    style: 'filter: sepia(.2) brightness(1.15) saturate(1.4)',
    beforeStyle: `background: rgba(125, 105, 24, .1); mix-blend-mode: multiply;`,
  }),

  defineFilter({
    name: 'amaro',
    style: 'filter: sepia(.35) contrast(1.1) brightness(1.2) saturate(1.3)',
    beforeStyle: `background: rgba(125, 105, 24, .2); mix-blend-mode: overlay;`,
  }),

  defineFilter({
    name: 'ashby',
    style: 'filter: sepia(.5) contrast(1.2) saturate(1.8)',
    beforeStyle: `background: rgba(125, 105, 24, .35); mix-blend-mode: lighten;`,
  }),

  defineFilter({
    name: 'brannan',
    style:
      'filter: sepia(.4) contrast(1.25) brightness(1.1) saturate(.9) hue-rotate(-2deg)',
  }),

  defineFilter({
    name: 'brooklyn',
    style: 'filter: sepia(.35) contrast(1.1) brightness(1.2) saturate(1.3)',
    beforeStyle: `background: rgba(127, 187, 227, .2); mix-blend-mode: overlay;`,
  }),

  defineFilter({
    name: 'charmes',
    style:
      'filter: sepia(.25) contrast(1.25) brightness(1.25) saturate(1.35) hue-rotate(-5deg)',
    beforeStyle: `background: rgba(125, 105, 24, .25); mix-blend-mode: darken;`,
  }),

  defineFilter({
    name: 'clarendon',
    style:
      'filter: sepia(.15) contrast(1.25) brightness(1.25) hue-rotate(5deg)',
    beforeStyle: `background: rgba(127, 187, 227, .4); mix-blend-mode: overlay;`,
  }),

  defineFilter({
    name: 'crema',
    style:
      'filter: sepia(.5) contrast(1.25) brightness(1.15) saturate(.9) hue-rotate(-2deg)',
    beforeStyle: `background: rgba(125, 105, 24, .2); mix-blend-mode: multiply;`,
  }),

  defineFilter({
    name: 'dogpatch',
    style: 'filter: sepia(.35) saturate(1.1) contrast(1.5)',
  }),

  defineFilter({
    name: 'earlybird',
    style:
      'filter: sepia(.25) contrast(1.25) brightness(1.15) saturate(.9) hue-rotate(-5deg)',
    beforeStyle: `background: radial-gradient(circle closest-corner, transparent 0, rgba(125, 105, 24, .2) 100%); mix-blend-mode: multiply;`,
  }),

  defineFilter({
    name: 'gingham',
    style: 'filter: contrast(1.1) brightness(1.1)',
    beforeStyle: `background: #e6e6e6; mix-blend-mode: soft-light;`,
  }),

  defineFilter({
    name: 'ginza',
    style:
      'filter: sepia(.25) contrast(1.15) brightness(1.2) saturate(1.35) hue-rotate(-5deg)',
    beforeStyle: `background: rgba(125, 105, 24, .15); mix-blend-mode: darken;`,
  }),

  defineFilter({
    name: 'hefe',
    style:
      'filter: sepia(.4) contrast(1.5) brightness(1.2) saturate(1.4) hue-rotate(-10deg)',
    beforeStyle: `background: radial-gradient(circle closest-corner, transparent 0, rgba(0, 0, 0, .25) 100%); mix-blend-mode: multiply;`,
  }),

  defineFilter({
    name: 'helena',
    style: 'filter: sepia(.5) contrast(1.05) brightness(1.05) saturate(1.35)',
    beforeStyle: `background: rgba(158, 175, 30, .25); mix-blend-mode: overlay;`,
  }),

  defineFilter({
    name: 'hudson',
    style:
      'filter: sepia(.25) contrast(1.2) brightness(1.2) saturate(1.05) hue-rotate(-15deg)',
    beforeStyle: `background: radial-gradient(circle closest-corner, transparent 25%, rgba(25, 62, 167, .25) 100%); mix-blend-mode: multiply;`,
  }),

  defineFilter({
    name: 'inkwell',
    style: 'filter: brightness(1.25) contrast(.85) grayscale(1)',
  }),

  defineFilter({
    name: 'juno',
    style: 'filter: sepia(.35) contrast(1.15) brightness(1.15) saturate(1.8)',
    beforeStyle: `background: rgba(127, 187, 227, .2); mix-blend-mode: overlay;`,
  }),

  defineFilter({
    name: 'kelvin',
    style:
      'filter: sepia(.15) contrast(1.5) brightness(1.1) hue-rotate(-10deg)',
    beforeStyle: `background: radial-gradient(circle closest-corner, rgba(128, 78, 15, .25) 0, rgba(128, 78, 15, .5) 100%); mix-blend-mode: overlay;`,
  }),

  defineFilter({
    name: 'lark',
    style: 'filter: sepia(.25) contrast(1.2) brightness(1.3) saturate(1.25)',
  }),

  defineFilter({
    name: 'lofi',
    style: 'filter: saturate(1.1) contrast(1.5)',
  }),

  defineFilter({
    name: 'ludwig',
    style: 'filter: sepia(.25) contrast(1.05) brightness(1.05) saturate(2)',
    beforeStyle: `background: rgba(125, 105, 24, .1); mix-blend-mode: overlay;`,
  }),

  defineFilter({
    name: 'maven',
    style: 'filter: sepia(.35) contrast(1.05) brightness(1.05) saturate(1.75)',
    beforeStyle: `background: rgba(158, 175, 30, .25); mix-blend-mode: darken;`,
  }),

  defineFilter({
    name: 'mayfair',
    style: 'filter: contrast(1.1) brightness(1.15) saturate(1.1)',
    beforeStyle: `background: radial-gradient(circle closest-corner, transparent 0, rgba(175, 105, 24, .4) 100%); mix-blend-mode: multiply;`,
  }),

  defineFilter({
    name: 'moon',
    style: 'filter: brightness(1.4) contrast(.95) saturate(0) sepia(.35)',
  }),

  defineFilter({
    name: 'nashville',
    style: 'filter: sepia(.25) contrast(1.5) brightness(.9) hue-rotate(-15deg)',
    beforeStyle: `background: radial-gradient(circle closest-corner, rgba(128, 78, 15, .5) 0, rgba(128, 78, 15, .65) 100%); mix-blend-mode: screen;`,
  }),

  defineFilter({
    name: 'perpetua',
    style: 'filter: contrast(1.1) brightness(1.25) saturate(1.1)',
    beforeStyle: `background: linear-gradient(to bottom, rgba(0, 91, 154, .25), rgba(230, 193, 61, .25)); mix-blend-mode: multiply;`,
  }),

  defineFilter({
    name: 'poprocket',
    style: 'filter: sepia(.15) brightness(1.2)',
    beforeStyle: `background: radial-gradient(circle closest-corner, rgba(206, 39, 70, .75) 40%, black 80%); mix-blend-mode: screen;`,
  }),

  defineFilter({
    name: 'reyes',
    style: 'filter: sepia(.75) contrast(.75) brightness(1.25) saturate(1.4)',
  }),

  defineFilter({
    name: 'rise',
    style: 'filter: sepia(.25) contrast(1.25) brightness(1.2) saturate(.9)',
    beforeStyle: `background: radial-gradient(circle closest-corner, transparent 0, rgba(230, 193, 61, .25) 100%); mix-blend-mode: lighten;`,
  }),

  defineFilter({
    name: 'sierra',
    style: 'filter: sepia(.25) contrast(1.5) brightness(.9) hue-rotate(-15deg)',
    beforeStyle: `background: radial-gradient(circle closest-corner, rgba(128, 78, 15, .5) 0, rgba(0, 0, 0, .65) 100%); mix-blend-mode: screen;`,
  }),

  defineFilter({
    name: 'skyline',
    style: 'filter: sepia(.15) contrast(1.25) brightness(1.25) saturate(1.2)',
  }),

  defineFilter({
    name: 'slumber',
    style: 'filter: sepia(.35) contrast(1.25) saturate(1.25)',
    beforeStyle: `background: rgba(125, 105, 24, .2); mix-blend-mode: darken;`,
  }),

  defineFilter({
    name: 'stinson',
    style: 'filter: sepia(.35) contrast(1.25) saturate(1.25)',
    beforeStyle: `background: rgba(125, 105, 24, .45); mix-blend-mode: lighten;`,
  }),

  defineFilter({
    name: 'sutro',
    style:
      'filter: sepia(.4) contrast(1.2) brightness(.9) saturate(1.4) hue-rotate(-10deg)',
    beforeStyle: `background: radial-gradient(circle closest-corner, transparent 50%, rgba(0, 0, 0, .5) 90%); mix-blend-mode: darken;`,
  }),

  defineFilter({
    name: 'toaster',
    style:
      'filter: sepia(.25) contrast(1.5) brightness(.95) hue-rotate(-15deg)',
    beforeStyle: `background: radial-gradient(circle, #804e0f, rgba(0, 0, 0, .25)); mix-blend-mode: screen;`,
  }),

  defineFilter({
    name: 'valencia',
    style: 'filter: sepia(.25) contrast(1.1) brightness(1.1)',
    beforeStyle: `background: rgba(230, 193, 61, .1); mix-blend-mode: lighten;`,
  }),

  defineFilter({
    name: 'vesper',
    style: 'filter: sepia(.35) contrast(1.15) brightness(1.2) saturate(1.3)',
    beforeStyle: `background: rgba(125, 105, 24, .25); mix-blend-mode: overlay;`,
  }),

  defineFilter({
    name: 'walden',
    style: 'filter: sepia(.35) contrast(.8) brightness(1.25) saturate(1.4)',
    beforeStyle: `background: rgba(229, 240, 128, .5); mix-blend-mode: darken;`,
  }),

  defineFilter({
    name: 'willow',
    style: 'filter: brightness(1.2) contrast(.85) saturate(.05) sepia(.2)',
  }),

  defineFilter({
    name: 'xpro-ii',
    style:
      'filter: sepia(.45) contrast(1.25) brightness(1.75) saturate(1.3) hue-rotate(-5deg)',
    beforeStyle: `background: radial-gradient(circle closest-corner, rgba(0, 91, 154, .35) 0, rgba(0, 0, 0, .65) 100%); mix-blend-mode: multiply;`,
  }),
]
