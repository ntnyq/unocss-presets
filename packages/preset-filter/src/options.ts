import type { PresetFilterOptions } from './types'

export const DEFAULT_OPTIONS: Required<PresetFilterOptions> = {
  prefix: 'filter-',
}

export function resolveOptions(
  options: PresetFilterOptions = {},
): Required<PresetFilterOptions> {
  return {
    ...DEFAULT_OPTIONS,
    ...options,
  }
}
