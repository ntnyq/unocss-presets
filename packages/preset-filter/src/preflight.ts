import type { Preflight } from '@unocss/core'
import type { PresetFilterOptions } from './types'

export const createPreflights = (options: Required<PresetFilterOptions>): Preflight[] => [
  {
    getCSS: () => `
      [class^="${options.prefix}"] {
        position: relative;
      }
      [class^="${options.prefix}"]:before {
        display: block;
        height: 100%;
        left: 0;
        position: absolute;
        top: 0;
        width: 100%;
        z-index: 1;
      }
      `,
  },
]
