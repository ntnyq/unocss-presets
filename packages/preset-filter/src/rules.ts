import { kebabCase } from 'uncase'
import { filters } from './filters'
import type { DynamicRule } from '@unocss/core'
import type { PresetFilterOptions } from './types'

export const createFilterRules = (options: Required<PresetFilterOptions>) => {
  const rules = filters.map<DynamicRule>(filter => [
    new RegExp(`^${options.prefix}${kebabCase(filter.name)}$`),
    () => [
      `.${options.prefix}${kebabCase(filter.name)} { ${filter.style} }`,
      filter.beforeStyle
        ? [
            `.${options.prefix}${kebabCase(filter.name)}:before { `,
            `  content: ""; ${filter.beforeStyle}`,
            ' }',
          ].join('\n')
        : '',
    ],
    {
      autocomplete: [`${options.prefix}${kebabCase(filter.name)}`],
    },
  ])
  return rules
}
