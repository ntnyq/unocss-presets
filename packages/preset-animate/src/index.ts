import { createPreflights } from './preflight'
import { createAnimationRules, createBaseRules } from './rules'
import type { Preset } from '@unocss/core'
import type { PresetAnimateOptions } from './types'

export const presetAnimate = (options: PresetAnimateOptions = {}): Preset => {
  options.preflight = options.preflight ?? true
  options.injectMediaQuery = options.injectMediaQuery ?? true
  options.variablePrefix = options.variablePrefix ?? 'un-animate-'
  options.extendAnimations = options.extendAnimations ?? []

  return {
    name: 'unocss-preset-animate',

    rules: [...createBaseRules(options), ...createAnimationRules(options)],

    preflights: options.preflight ? createPreflights(options) : [],
  }
}

export * from './types'
export * from './animations'

export default presetAnimate
