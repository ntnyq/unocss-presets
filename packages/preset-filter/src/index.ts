import { resolveOptions } from './options'
import { createPreflights } from './preflight'
import { createFilterRules } from './rules'
import type { Preset } from '@unocss/core'
import type { PresetFilterOptions } from './types'

export const presetFilter = (options: Partial<PresetFilterOptions> = {}): Preset => {
  const resolvedOptions = resolveOptions(options)
  return {
    name: 'unocss-preset-filter',
    rules: [...createFilterRules(resolvedOptions)],
    preflights: createPreflights(resolvedOptions),
  }
}

export * from './types'
export * from './options'
export * from './filters'

export default presetFilter
