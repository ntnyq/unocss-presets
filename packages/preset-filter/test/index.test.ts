import { kebabCase } from '@ntnyq/utils'
import { createGenerator } from '@unocss/core'
import { describe, expect, it } from 'vitest'
import { filters, presetFilter } from '../src'

describe('selectors', () => {
  it('base selectors', async () => {
    const inputs = filters.map(filter => `filter-${kebabCase(filter.name)}`)
    const generator = await createGenerator({
      presets: [presetFilter()],
    })
    const { css } = await generator.generate(inputs.join('\n'))
    expect(css).toMatchSnapshot()
  })

  it('selectors when custom prefix', async () => {
    const inputs = filters.map(filter => `f-${kebabCase(filter.name)}`)
    const generator = await createGenerator({
      presets: [presetFilter({ prefix: 'f-' })],
    })
    const { css } = await generator.generate(inputs.join('\n'))
    expect(css).toMatchSnapshot()
  })
})
