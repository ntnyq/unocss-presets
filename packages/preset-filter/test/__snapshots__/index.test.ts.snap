// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`selectors > base selectors 1`] = `
"/* layer: preflights */
[class^="filter-"] {
  position: relative;
}
[class^="filter-"]:before {
  display: block;
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  width: 100%;
  z-index: 1;
}
/* layer: default */
.filter-1977 { filter: sepia(.5) hue-rotate(-30deg) saturate(1.4) }
.filter-aden { filter: sepia(.2) brightness(1.15) saturate(1.4) }
.filter-aden:before { 
  content: ""; background: rgba(125, 105, 24, .1); mix-blend-mode: multiply;
 }
.filter-amaro { filter: sepia(.35) contrast(1.1) brightness(1.2) saturate(1.3) }
.filter-amaro:before { 
  content: ""; background: rgba(125, 105, 24, .2); mix-blend-mode: overlay;
 }
.filter-ashby { filter: sepia(.5) contrast(1.2) saturate(1.8) }
.filter-ashby:before { 
  content: ""; background: rgba(125, 105, 24, .35); mix-blend-mode: lighten;
 }
.filter-brannan { filter: sepia(.4) contrast(1.25) brightness(1.1) saturate(.9) hue-rotate(-2deg) }
.filter-brooklyn { filter: sepia(.35) contrast(1.1) brightness(1.2) saturate(1.3) }
.filter-brooklyn:before { 
  content: ""; background: rgba(127, 187, 227, .2); mix-blend-mode: overlay;
 }
.filter-charmes { filter: sepia(.25) contrast(1.25) brightness(1.25) saturate(1.35) hue-rotate(-5deg) }
.filter-charmes:before { 
  content: ""; background: rgba(125, 105, 24, .25); mix-blend-mode: darken;
 }
.filter-clarendon { filter: sepia(.15) contrast(1.25) brightness(1.25) hue-rotate(5deg) }
.filter-clarendon:before { 
  content: ""; background: rgba(127, 187, 227, .4); mix-blend-mode: overlay;
 }
.filter-crema { filter: sepia(.5) contrast(1.25) brightness(1.15) saturate(.9) hue-rotate(-2deg) }
.filter-crema:before { 
  content: ""; background: rgba(125, 105, 24, .2); mix-blend-mode: multiply;
 }
.filter-dogpatch { filter: sepia(.35) saturate(1.1) contrast(1.5) }
.filter-earlybird { filter: sepia(.25) contrast(1.25) brightness(1.15) saturate(.9) hue-rotate(-5deg) }
.filter-earlybird:before { 
  content: ""; background: radial-gradient(circle closest-corner, transparent 0, rgba(125, 105, 24, .2) 100%); mix-blend-mode: multiply;
 }
.filter-gingham { filter: contrast(1.1) brightness(1.1) }
.filter-gingham:before { 
  content: ""; background: #e6e6e6; mix-blend-mode: soft-light;
 }
.filter-ginza { filter: sepia(.25) contrast(1.15) brightness(1.2) saturate(1.35) hue-rotate(-5deg) }
.filter-ginza:before { 
  content: ""; background: rgba(125, 105, 24, .15); mix-blend-mode: darken;
 }
.filter-hefe { filter: sepia(.4) contrast(1.5) brightness(1.2) saturate(1.4) hue-rotate(-10deg) }
.filter-hefe:before { 
  content: ""; background: radial-gradient(circle closest-corner, transparent 0, rgba(0, 0, 0, .25) 100%); mix-blend-mode: multiply;
 }
.filter-helena { filter: sepia(.5) contrast(1.05) brightness(1.05) saturate(1.35) }
.filter-helena:before { 
  content: ""; background: rgba(158, 175, 30, .25); mix-blend-mode: overlay;
 }
.filter-hudson { filter: sepia(.25) contrast(1.2) brightness(1.2) saturate(1.05) hue-rotate(-15deg) }
.filter-hudson:before { 
  content: ""; background: radial-gradient(circle closest-corner, transparent 25%, rgba(25, 62, 167, .25) 100%); mix-blend-mode: multiply;
 }
.filter-inkwell { filter: brightness(1.25) contrast(.85) grayscale(1) }
.filter-juno { filter: sepia(.35) contrast(1.15) brightness(1.15) saturate(1.8) }
.filter-juno:before { 
  content: ""; background: rgba(127, 187, 227, .2); mix-blend-mode: overlay;
 }
.filter-kelvin { filter: sepia(.15) contrast(1.5) brightness(1.1) hue-rotate(-10deg) }
.filter-kelvin:before { 
  content: ""; background: radial-gradient(circle closest-corner, rgba(128, 78, 15, .25) 0, rgba(128, 78, 15, .5) 100%); mix-blend-mode: overlay;
 }
.filter-lark { filter: sepia(.25) contrast(1.2) brightness(1.3) saturate(1.25) }
.filter-lofi { filter: saturate(1.1) contrast(1.5) }
.filter-ludwig { filter: sepia(.25) contrast(1.05) brightness(1.05) saturate(2) }
.filter-ludwig:before { 
  content: ""; background: rgba(125, 105, 24, .1); mix-blend-mode: overlay;
 }
.filter-maven { filter: sepia(.35) contrast(1.05) brightness(1.05) saturate(1.75) }
.filter-maven:before { 
  content: ""; background: rgba(158, 175, 30, .25); mix-blend-mode: darken;
 }
.filter-mayfair { filter: contrast(1.1) brightness(1.15) saturate(1.1) }
.filter-mayfair:before { 
  content: ""; background: radial-gradient(circle closest-corner, transparent 0, rgba(175, 105, 24, .4) 100%); mix-blend-mode: multiply;
 }
.filter-moon { filter: brightness(1.4) contrast(.95) saturate(0) sepia(.35) }
.filter-nashville { filter: sepia(.25) contrast(1.5) brightness(.9) hue-rotate(-15deg) }
.filter-nashville:before { 
  content: ""; background: radial-gradient(circle closest-corner, rgba(128, 78, 15, .5) 0, rgba(128, 78, 15, .65) 100%); mix-blend-mode: screen;
 }
.filter-perpetua { filter: contrast(1.1) brightness(1.25) saturate(1.1) }
.filter-perpetua:before { 
  content: ""; background: linear-gradient(to bottom, rgba(0, 91, 154, .25), rgba(230, 193, 61, .25)); mix-blend-mode: multiply;
 }
.filter-poprocket { filter: sepia(.15) brightness(1.2) }
.filter-poprocket:before { 
  content: ""; background: radial-gradient(circle closest-corner, rgba(206, 39, 70, .75) 40%, black 80%); mix-blend-mode: screen;
 }
.filter-reyes { filter: sepia(.75) contrast(.75) brightness(1.25) saturate(1.4) }
.filter-rise { filter: sepia(.25) contrast(1.25) brightness(1.2) saturate(.9) }
.filter-rise:before { 
  content: ""; background: radial-gradient(circle closest-corner, transparent 0, rgba(230, 193, 61, .25) 100%); mix-blend-mode: lighten;
 }
.filter-sierra { filter: sepia(.25) contrast(1.5) brightness(.9) hue-rotate(-15deg) }
.filter-sierra:before { 
  content: ""; background: radial-gradient(circle closest-corner, rgba(128, 78, 15, .5) 0, rgba(0, 0, 0, .65) 100%); mix-blend-mode: screen;
 }
.filter-skyline { filter: sepia(.15) contrast(1.25) brightness(1.25) saturate(1.2) }
.filter-slumber { filter: sepia(.35) contrast(1.25) saturate(1.25) }
.filter-slumber:before { 
  content: ""; background: rgba(125, 105, 24, .2); mix-blend-mode: darken;
 }
.filter-stinson { filter: sepia(.35) contrast(1.25) saturate(1.25) }
.filter-stinson:before { 
  content: ""; background: rgba(125, 105, 24, .45); mix-blend-mode: lighten;
 }
.filter-sutro { filter: sepia(.4) contrast(1.2) brightness(.9) saturate(1.4) hue-rotate(-10deg) }
.filter-sutro:before { 
  content: ""; background: radial-gradient(circle closest-corner, transparent 50%, rgba(0, 0, 0, .5) 90%); mix-blend-mode: darken;
 }
.filter-toaster { filter: sepia(.25) contrast(1.5) brightness(.95) hue-rotate(-15deg) }
.filter-toaster:before { 
  content: ""; background: radial-gradient(circle, #804e0f, rgba(0, 0, 0, .25)); mix-blend-mode: screen;
 }
.filter-valencia { filter: sepia(.25) contrast(1.1) brightness(1.1) }
.filter-valencia:before { 
  content: ""; background: rgba(230, 193, 61, .1); mix-blend-mode: lighten;
 }
.filter-vesper { filter: sepia(.35) contrast(1.15) brightness(1.2) saturate(1.3) }
.filter-vesper:before { 
  content: ""; background: rgba(125, 105, 24, .25); mix-blend-mode: overlay;
 }
.filter-walden { filter: sepia(.35) contrast(.8) brightness(1.25) saturate(1.4) }
.filter-walden:before { 
  content: ""; background: rgba(229, 240, 128, .5); mix-blend-mode: darken;
 }
.filter-willow { filter: brightness(1.2) contrast(.85) saturate(.05) sepia(.2) }
.filter-xpro-ii { filter: sepia(.45) contrast(1.25) brightness(1.75) saturate(1.3) hue-rotate(-5deg) }
.filter-xpro-ii:before { 
  content: ""; background: radial-gradient(circle closest-corner, rgba(0, 91, 154, .35) 0, rgba(0, 0, 0, .65) 100%); mix-blend-mode: multiply;
 }"
`;

exports[`selectors > selectors when custom prefix 1`] = `
"/* layer: preflights */
[class^="f-"] {
  position: relative;
}
[class^="f-"]:before {
  display: block;
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  width: 100%;
  z-index: 1;
}
/* layer: default */
.f-1977 { filter: sepia(.5) hue-rotate(-30deg) saturate(1.4) }
.f-aden { filter: sepia(.2) brightness(1.15) saturate(1.4) }
.f-aden:before { 
  content: ""; background: rgba(125, 105, 24, .1); mix-blend-mode: multiply;
 }
.f-amaro { filter: sepia(.35) contrast(1.1) brightness(1.2) saturate(1.3) }
.f-amaro:before { 
  content: ""; background: rgba(125, 105, 24, .2); mix-blend-mode: overlay;
 }
.f-ashby { filter: sepia(.5) contrast(1.2) saturate(1.8) }
.f-ashby:before { 
  content: ""; background: rgba(125, 105, 24, .35); mix-blend-mode: lighten;
 }
.f-brannan { filter: sepia(.4) contrast(1.25) brightness(1.1) saturate(.9) hue-rotate(-2deg) }
.f-brooklyn { filter: sepia(.35) contrast(1.1) brightness(1.2) saturate(1.3) }
.f-brooklyn:before { 
  content: ""; background: rgba(127, 187, 227, .2); mix-blend-mode: overlay;
 }
.f-charmes { filter: sepia(.25) contrast(1.25) brightness(1.25) saturate(1.35) hue-rotate(-5deg) }
.f-charmes:before { 
  content: ""; background: rgba(125, 105, 24, .25); mix-blend-mode: darken;
 }
.f-clarendon { filter: sepia(.15) contrast(1.25) brightness(1.25) hue-rotate(5deg) }
.f-clarendon:before { 
  content: ""; background: rgba(127, 187, 227, .4); mix-blend-mode: overlay;
 }
.f-crema { filter: sepia(.5) contrast(1.25) brightness(1.15) saturate(.9) hue-rotate(-2deg) }
.f-crema:before { 
  content: ""; background: rgba(125, 105, 24, .2); mix-blend-mode: multiply;
 }
.f-dogpatch { filter: sepia(.35) saturate(1.1) contrast(1.5) }
.f-earlybird { filter: sepia(.25) contrast(1.25) brightness(1.15) saturate(.9) hue-rotate(-5deg) }
.f-earlybird:before { 
  content: ""; background: radial-gradient(circle closest-corner, transparent 0, rgba(125, 105, 24, .2) 100%); mix-blend-mode: multiply;
 }
.f-gingham { filter: contrast(1.1) brightness(1.1) }
.f-gingham:before { 
  content: ""; background: #e6e6e6; mix-blend-mode: soft-light;
 }
.f-ginza { filter: sepia(.25) contrast(1.15) brightness(1.2) saturate(1.35) hue-rotate(-5deg) }
.f-ginza:before { 
  content: ""; background: rgba(125, 105, 24, .15); mix-blend-mode: darken;
 }
.f-hefe { filter: sepia(.4) contrast(1.5) brightness(1.2) saturate(1.4) hue-rotate(-10deg) }
.f-hefe:before { 
  content: ""; background: radial-gradient(circle closest-corner, transparent 0, rgba(0, 0, 0, .25) 100%); mix-blend-mode: multiply;
 }
.f-helena { filter: sepia(.5) contrast(1.05) brightness(1.05) saturate(1.35) }
.f-helena:before { 
  content: ""; background: rgba(158, 175, 30, .25); mix-blend-mode: overlay;
 }
.f-hudson { filter: sepia(.25) contrast(1.2) brightness(1.2) saturate(1.05) hue-rotate(-15deg) }
.f-hudson:before { 
  content: ""; background: radial-gradient(circle closest-corner, transparent 25%, rgba(25, 62, 167, .25) 100%); mix-blend-mode: multiply;
 }
.f-inkwell { filter: brightness(1.25) contrast(.85) grayscale(1) }
.f-juno { filter: sepia(.35) contrast(1.15) brightness(1.15) saturate(1.8) }
.f-juno:before { 
  content: ""; background: rgba(127, 187, 227, .2); mix-blend-mode: overlay;
 }
.f-kelvin { filter: sepia(.15) contrast(1.5) brightness(1.1) hue-rotate(-10deg) }
.f-kelvin:before { 
  content: ""; background: radial-gradient(circle closest-corner, rgba(128, 78, 15, .25) 0, rgba(128, 78, 15, .5) 100%); mix-blend-mode: overlay;
 }
.f-lark { filter: sepia(.25) contrast(1.2) brightness(1.3) saturate(1.25) }
.f-lofi { filter: saturate(1.1) contrast(1.5) }
.f-ludwig { filter: sepia(.25) contrast(1.05) brightness(1.05) saturate(2) }
.f-ludwig:before { 
  content: ""; background: rgba(125, 105, 24, .1); mix-blend-mode: overlay;
 }
.f-maven { filter: sepia(.35) contrast(1.05) brightness(1.05) saturate(1.75) }
.f-maven:before { 
  content: ""; background: rgba(158, 175, 30, .25); mix-blend-mode: darken;
 }
.f-mayfair { filter: contrast(1.1) brightness(1.15) saturate(1.1) }
.f-mayfair:before { 
  content: ""; background: radial-gradient(circle closest-corner, transparent 0, rgba(175, 105, 24, .4) 100%); mix-blend-mode: multiply;
 }
.f-moon { filter: brightness(1.4) contrast(.95) saturate(0) sepia(.35) }
.f-nashville { filter: sepia(.25) contrast(1.5) brightness(.9) hue-rotate(-15deg) }
.f-nashville:before { 
  content: ""; background: radial-gradient(circle closest-corner, rgba(128, 78, 15, .5) 0, rgba(128, 78, 15, .65) 100%); mix-blend-mode: screen;
 }
.f-perpetua { filter: contrast(1.1) brightness(1.25) saturate(1.1) }
.f-perpetua:before { 
  content: ""; background: linear-gradient(to bottom, rgba(0, 91, 154, .25), rgba(230, 193, 61, .25)); mix-blend-mode: multiply;
 }
.f-poprocket { filter: sepia(.15) brightness(1.2) }
.f-poprocket:before { 
  content: ""; background: radial-gradient(circle closest-corner, rgba(206, 39, 70, .75) 40%, black 80%); mix-blend-mode: screen;
 }
.f-reyes { filter: sepia(.75) contrast(.75) brightness(1.25) saturate(1.4) }
.f-rise { filter: sepia(.25) contrast(1.25) brightness(1.2) saturate(.9) }
.f-rise:before { 
  content: ""; background: radial-gradient(circle closest-corner, transparent 0, rgba(230, 193, 61, .25) 100%); mix-blend-mode: lighten;
 }
.f-sierra { filter: sepia(.25) contrast(1.5) brightness(.9) hue-rotate(-15deg) }
.f-sierra:before { 
  content: ""; background: radial-gradient(circle closest-corner, rgba(128, 78, 15, .5) 0, rgba(0, 0, 0, .65) 100%); mix-blend-mode: screen;
 }
.f-skyline { filter: sepia(.15) contrast(1.25) brightness(1.25) saturate(1.2) }
.f-slumber { filter: sepia(.35) contrast(1.25) saturate(1.25) }
.f-slumber:before { 
  content: ""; background: rgba(125, 105, 24, .2); mix-blend-mode: darken;
 }
.f-stinson { filter: sepia(.35) contrast(1.25) saturate(1.25) }
.f-stinson:before { 
  content: ""; background: rgba(125, 105, 24, .45); mix-blend-mode: lighten;
 }
.f-sutro { filter: sepia(.4) contrast(1.2) brightness(.9) saturate(1.4) hue-rotate(-10deg) }
.f-sutro:before { 
  content: ""; background: radial-gradient(circle closest-corner, transparent 50%, rgba(0, 0, 0, .5) 90%); mix-blend-mode: darken;
 }
.f-toaster { filter: sepia(.25) contrast(1.5) brightness(.95) hue-rotate(-15deg) }
.f-toaster:before { 
  content: ""; background: radial-gradient(circle, #804e0f, rgba(0, 0, 0, .25)); mix-blend-mode: screen;
 }
.f-valencia { filter: sepia(.25) contrast(1.1) brightness(1.1) }
.f-valencia:before { 
  content: ""; background: rgba(230, 193, 61, .1); mix-blend-mode: lighten;
 }
.f-vesper { filter: sepia(.35) contrast(1.15) brightness(1.2) saturate(1.3) }
.f-vesper:before { 
  content: ""; background: rgba(125, 105, 24, .25); mix-blend-mode: overlay;
 }
.f-walden { filter: sepia(.35) contrast(.8) brightness(1.25) saturate(1.4) }
.f-walden:before { 
  content: ""; background: rgba(229, 240, 128, .5); mix-blend-mode: darken;
 }
.f-willow { filter: brightness(1.2) contrast(.85) saturate(.05) sepia(.2) }
.f-xpro-ii { filter: sepia(.45) contrast(1.25) brightness(1.75) saturate(1.3) hue-rotate(-5deg) }
.f-xpro-ii:before { 
  content: ""; background: radial-gradient(circle closest-corner, rgba(0, 91, 154, .35) 0, rgba(0, 0, 0, .65) 100%); mix-blend-mode: multiply;
 }"
`;
